src/components/Heading.tsx

```diff
@@ -0,0 +1,46 @@
+import { FunctionComponent, ReactNode } from 'react'
+
+import classNames from 'classnames'
+
+interface Heading {
+    as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
+    size: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
+    children: ReactNode | string
+    className?: string
+}
+
+/**
+ * This component renders our typography headings as defined in our DLS. This
+ * allows us to render specifically sized headings with semantic and valid
+ * ordering. Heading elements should always be in sequentially descending order
+ * for accessibility and valid page structure.
+ *
+ * @param props - component props
+ * @param props.as - the type of heading to render regardless of size. This is important
+ * to avoid skipping heading levels for valid ordering.
+ * @param props.size - the size of heading
+ * @param props.children - the children of the heading
+ * @param props.className - the className of the heading
+ */
+export const Heading: FunctionComponent<Heading> = ({ as, size, children, className }) => {
```

Comment:
This is great! Takes care of some headings error that the ARC Toolkit test highlights ([ref](https://docs.sourcegraph.com/dev/background-information/web/accessibility/detailed-checklist))